nomos-libp2p/src/lib.rs

```diff
@@ -79,11 +182,34 @@ impl Swarm {
             .with_tokio()
             .with_quic()
             .with_dns()?
-            .with_behaviour(|_| Behaviour::new(peer_id, config.gossipsub_config.clone()).unwrap())?
+            .with_behaviour(|keypair| {
+                Behaviour::new(
+                    peer_id,
+                    config.gossipsub_config.clone(),
+                    config.kademlia_config.clone(),
+                    config.identify_config.clone(),
```

Comment:
Nitpik: probably no need to clone. Destructure the config and just pass the configs so ownership is splitted.

Reply:
Should be fixed now. Needed to change parent configs to be moved instead of borrowed.